gemrb/core/System/swab.h

```diff
@@ -30,7 +30,13 @@ extern "C" {
 #endif
 
 #ifndef HAVE_SWAB
+
+#ifdef HAVE_CSWAB
+#define swab(a,b,c)  swab((char*)(a), (char*)(b), (c))
```

Comment:
lets use `static_cast` to avoid more sonar complaints

Reply:
actually... I see we also want to cast away const. im unsure if that is wise here. how do we know the implementation isn't writing back to a? that would be bad for us. it would be better then to just expose our `swap_private` as some other name and use it instead.

Reply:
Yeah, all C++-casts I tried ended up in complaints. No objections, drop this shit alltogether.